### PR TITLE
Ubuntu 24.04: Implement 5.3.2.4 Ensure pam_pwhistory module is enabled

### PR DESCRIPTION
**Pull request by alanmcanonical**
#### Description:

- Implement 5.3.2.4 Ensure pam_pwhistory module is enabled
- Rule checks that pam_pwhistory.so is enabled in /etc/pam.d/common-password
- Implement bash_pam_pwhistory_enable macro

#### Rationale:

- Satisfies Ubuntu 24.04 CIS control 5.3.2.4

### CODE REVIEW

File: shared/macros/10-bash.jinja [hunk 1]
```diff
@@ -982,7 +982,24 @@ if [ -f /usr/bin/authselect ]; then
...
+conf_name={{{ pam_file }}}
+conf_path="/usr/share/pam-configs"
+
+if [ ! -f "$conf_path"/"$conf_name" ]; then
+    cat << EOF > "$conf_path"/"$conf_name"
+Name: pwhistory password history checking
+Default: yes
+Priority: 1024
+Password-Type: Primary
+Password: {{{ control }}} pam_pwhistory.so remember=24 enforce_for_root try_first_pass use_authtok
```
Comment (mpurg) [inline shared/macros/10-bash.jinja:995]: Should this be one line or
```
Password:
    {{{ control }}} pam_pwhistory.so remember=24 enforce_for_root try_first_pass use_authtok
```

**Review by mpurg — CHANGES_REQUESTED** [review]
Minor change

File: shared/macros/10-bash.jinja (hunk 1 above)
Comment (alanmcanonical) [reply shared/macros/10-bash.jinja:995]: pam-auth-update can identify the non new-line conf, so no need to change this

**Review by dodys — APPROVED** [review]
lgtm, thanks!

**Pull request merged**